new/tk-mixlib-install-unification.md

```diff
@@ -0,0 +1,88 @@
+---
+RFC: unassigned
+Title: Test Kitchen Mixlib Install Unification
+Author: Patrick Wright <patrick@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Test Kitchen Mixlib Install Unification
+
+Test Kitchen currently supports two code paths for configuring, downloading and installing chef and chefdk on test instances. Both code paths use Mixlib Install but are implemented differently. In the past year, significant effort has been put into Mixlib Install to make it the canonical source for which all Chef related products are queried for meta-data and for generating the bootstrap installation scripts (install.sh and install.ps1). 
+
+Mixlib Install's new API is currently used for downloads.chef.io and omnitruck.chef.io. It's also used for the chef-ingredient cookbook. Chef transparently implemented the new API and selected config options in Kitchen for testing purposes. It has proven valuable and easy to use. The feature set is limited at this time.
+
+## Motivation
+
+```
+As a Kitchen User,
+I want to use simple, clear configuration options,
+so that I can setup Kitchen config files with less complexity and obfuscation.
+```
+
+```
+As a Kitchen Developer,
+I want to maintain a single Mixlib Install code path,
+so that I can benefit from the latest API changes and reduce maintenance risk.
+```
+
+The current Mixlib Install path used in Kitchen was developed specifically for Kitchen and is tightly coupled with Kitchen's provisioner options. It's referred to as the `ScriptGenerator`. The new API was developed to be used by any project that needs to query Chef product meta-data or generate an installation script. The new API takes into account the specifics of interacting with Chef's distribution infrastructure (Package Router).
+
+Two feature requests have been made which lend themselves to the new API. The first is
+caching
+user agent
+
+## Specification
+
+This work is separated into 3 phases.
+
+##### Phase 1 - Incrementally add new transparent configuration options
+
+* All existing configuration options behave as expected
+* Build feature parity with the new options to match existing functionality
+* Setting `product_name` will trigger the new options use cases
+* Maintain both Mixlib Install code paths
+* Document new options in the Kitchen repo
+
+##### Phase 2 - Provide option deprecation warnings and "how-tos"
+
+* All existing configuration options still behave as expected
+* User is clearly warned when using options that will be deprecated
+* Deprecation warnings include explanations and resolution instructions
+
+##### Phase 3 - Make the switch
+
+* Raise errors along with instructions when using deprecated options 
+* `ScriptGenerator` code path removed from Kitchen
+* docs.chef.io and kitchen.ci documentation updates
+
+### Phase 1 Details
+
+Here is an example of how the new options `product_version` and `install_strategy` replace `require_chef_omnibus`: https://gist.github.com/wrightp/60f8723b48d80de9b1ed0fc028757e32
```

Comment:
Perhaps we should just bring the gist contents into this document? That would make things more "official" and ensure this document becomes the canonical source of what is changing.